core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -247,10 +252,14 @@ private void executeFateOperation(long opid, FateOperation op, List<ByteBuffer>
       try {
         client = MasterClient.getConnectionWithRetry(context);
         client.executeFateOperation(Tracer.traceInfo(), context.rpcCreds(), opid, op, args, opts, autoCleanUp);
-        break;
+        return;
       } catch (TTransportException tte) {
         log.debug("Failed to call executeFateOperation(), retrying ... ", tte);
         sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
+      } catch (ThriftNotActiveServiceException e) {
```

Comment:
could almost use the new java convention where catch can list multiple exceptions, except log msgs differ